languages/typescript/src/main/kotlin/io/vrap/codegen/languages/typescript/joi/JoiObjectTypeExtensions.kt

```diff
@@ -1,22 +1,28 @@
 package io.vrap.codegen.languages.typescript.joi
 
-
-import io.vrap.codegen.languages.php.extensions.simpleName
+import io.vrap.codegen.languages.extensions.ExtensionsBase
+import io.vrap.codegen.languages.typescript.model.simpleTSName
 import io.vrap.rmf.codegen.types.*
 import io.vrap.rmf.raml.model.types.AnyType
 import io.vrap.rmf.raml.model.types.ObjectType
-import java.lang.IllegalStateException
+import io.vrap.rmf.raml.model.types.UnionType
 import java.nio.file.Path
 import java.nio.file.Paths
 
-interface JoiObjectTypeExtensions : io.vrap.codegen.languages.ExtensionsBase {
+interface JoiObjectTypeExtensions : ExtensionsBase {
 
     fun getImportsForJoiModule(moduleName: String, types: List<AnyType>): String {
 
         return types
-                .filter { it is ObjectType }
-                .map { it as ObjectType }
-                .flatMap { it.getDependencies() }
+                .filter { it is ObjectType || it is UnionType}
```

Comment:
Do we really use union types?

Reply:
in the official spec we have had an attribute case which is a union, since it dosn't take a lot of effort to support it i added it.